src/ppl/src/IOPlacer.cpp

```diff
@@ -1486,15 +1492,24 @@ bool IOPlacer::assignPinToSection(IOPin& io_pin,
                                   std::vector<Section>& sections)
 {
   bool pin_assigned = false;
+  bool has_mirrored_pin = io_pin.getBTerm()->hasMirroredBTerm();
 
   if (!io_pin.isInGroup() && !io_pin.isAssignedToSection()
       && !io_pin.inFallback()) {
     std::vector<int> dst(sections.size());
+    std::vector<int> larger_cost(sections.size());
+
     for (int i = 0; i < sections.size(); i++) {
       dst[i] = netlist_->computeIONetHPWL(idx, sections[i].pos)
                + getMirroredPinCost(io_pin, sections[i].pos);
+
+      if (has_mirrored_pin) {
+        larger_cost[i]
+            = std::max(netlist_->computeIONetHPWL(idx, sections[i].pos),
+                       getMirroredPinCost(io_pin, sections[i].pos));
```

Comment:
You just computed both of these values above.  Let's avoid doing the work twice.